File: 백준/Gold/11657. 타임머신/타임머신.py
import sys
input = sys.stdin.readline

n, m = map(int, input().split())
graph = []
for _ in range(m):
    a, b, c = map(int, input().split())
    graph.append((a, b, c))

INF = sys.maxsize
def bellmanford(start):
    distance = [INF] * (n+1)
    distance[start] = 0

    for i in range(n):
        for j in range(m):
            cur_node, next_node, cur_weight = graph[j]
            if distance[cur_node] == INF:
                continue
            if distance[cur_node] + cur_weight < distance[next_node]:
                distance[next_node] = distance[cur_node] + cur_weight
                if i == n - 1:
                    return [-1]
    
    for i in range(n+1):
        if distance[i] == INF:
            distance[i] = -1
    
    return distance[2:]

print(*bellmanford(1), sep='\n')

File: 백준/Gold/11657. 타임머신/타임머신.js
const input = require('fs').readFileSync(process.platform === 'linux' ? '/dev/stdin' : __dirname + '/input.txt').toString().split('\n')

const [N, M] = input[0].split(' ').map(Number)
const bus = []
for (let i = 1; i < M + 1; i++) {
  const [A, B, C] = input[i].split(' ').map(Number)
  bus.push([A, B, C])
}
console.log(bellman_ford())

function bellman_ford() {
  const distance = Array(N + 1).fill(Infinity)
  distance[1] = 0
  
  for (let i = 0; i < N; i++) {
    for (const [A, B, C] of bus) {
      if (distance[A] === Infinity) continue
      if (distance[B] <= distance[A] + C) continue
      
      distance[B] = distance[A] + C
      
      if (i === N - 1) return - 1
    }
  }

  return distance.slice(2).map((d) => d === Infinity ? -1 : d).join('\n')
}